poisson_lib.py

```
@@ -128,647 +163,625 @@ def analyze_poisson_outcome(
+        offset = None
+        if offset_col and offset_col in df_aligned.columns:
+            offset = np.log(pd.to_numeric(df_aligned[offset_col], errors='coerce').fillna(1))
+            logger.info(f"Using offset column: {offset_col}")
+        
+        candidates = []
+        results_db = {}
+        sorted_cols = sorted(df.columns.astype(str))
+        mode_map = {}
+        cat_levels_map = {}
+        
+        def fmt_p(val: Union[float, str]) -> str:
```

Ruff (0.14.10)

Use `X | Y` for type annotations

Convert to `X | Y`

(UP007)